src/profile-logic/tracks.js

```diff
@@ -110,6 +110,27 @@ function _getDefaultLocalTrackOrder(tracks: LocalTrack[]) {
       return 1;
     }
 
+    // If the tracks are both threads, sort them by thread name, and then by
+    // creation time if they have the same name.
+    if (tracks[a].type === 'thread' && tracks[b].type === 'thread' && profile) {
+      const idxA = tracks[a].threadIndex;
+      const idxB = tracks[b].threadIndex;
+      if (!idxA || !idxB) {
```

Comment:
Note that thread indexes can be 0 :) so you should check against undefined instead.
But I believe this check isn't necessary as we're guaranteed that `threadIndex` is present when the type is `thread`. I double checked locally and it looks like Flow is happy with this condition removed.